realm/realm-library/src/androidTest/java/io/realm/RealmAsyncQueryTests.java

```diff
@@ -458,6 +454,17 @@ public void onSuccess() {
                 looperThread.testComplete();
             }
         });
+
+        // Wait for all async tasks finish to ensure the async transaction posted callback will arrive first.
+        TestHelper.resetRealmThreadExecutor();
```

Comment:
try...finally will not fix it, the exception will still be thrown, but it will ensure we are not breaking 100 other tests.